### PR TITLE
upgrade rules_nixpkgs

### PR DESCRIPTION
**Pull request by aherrmann-da**
Upgrade rules nixpkgs which now includes the hermetic cc toolchain

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -102,11 +102,7 @@ common_nix_file_deps = [
 ]
 
 # Use Nix provisioned cc toolchain
-nixpkgs_cc_configure_hermetic(
-    # We override the Bazel's autodetect toolchain to avoid accidentaly
-    # dependencies on the inhermetic autodetected builtin include paths or
-    # builds failing due to Bazel not finding `cc` in `$PATH` or `$CC`.
-    name = "local_config_cc",
```
Comment (aherrmann-da) [inline WORKSPACE:109]: This name is now the default value.

File: bazel_tools/nixpkgs-disable-http2.patch [hunk 2]
```diff
@@ -11,12 +11,3 @@ index 2f2c83d..61123c6 100644
      if repository_ctx.attr.nix_file and repository_ctx.attr.nix_file_content:
          fail("Specify one of 'nix_file' or 'nix_file_content', but not both.")
      elif repository_ctx.attr.nix_file:
-@@ -186,6 +186,8 @@ def _nixpkgs_package_impl(repository_ctx):
-         "bazel-support/nix-out-link",
-     ])
- 
-+    expr_args.extend(["--option", "http2", "false"])
```
Comment (cocreature) [inline bazel_tools/nixpkgs-disable-http2.patch:18]: Why is this unnecessary now?

**Review by cocreature — APPROVED** [review]
Nice, thanks! One question on the http2 patch.

File: bazel_tools/nixpkgs-disable-http2.patch (hunk 2 above)
Comment (aherrmann-da) [reply bazel_tools/nixpkgs-disable-http2.patch:18]: It's duplicated further up in [line 10](https://github.com/digital-asset/daml/blob/d16a4b3b3cbdbf9f820585467ee594409b0d41b6/bazel_tools/nixpkgs-disable-http2.patch#L10). So, the flag is still there.

**Pull request merged**